src/Components/test/E2ETest/Tests/InputLargeTextAreaTest.cs

```diff
@@ -0,0 +1,208 @@
+            SetTextAreaValueInBrowser('a');
+
+            getTextBtn.Click();
+            Browser.Equal(newValue, () => textResultFromComponent.GetAttribute("innerHTML"));
+
+            FocusAway();
+            AssertLogDoesNotContainMessages(CircuitErrors);
+        }
+
+        [Fact]
+        public void CanGetValue_ThrowsIfTextAreaHasMoreContentThanMaxAllowed()
+        {
+            // Prime the textarea with 50k chars (more than 32k default limit)
+            SetTextAreaValueInBrowser('a');
+
+            var getLimitedTextBtn = Browser.Exists(By.Id("getLimitedTextBtn"));
+            getLimitedTextBtn.Click();
+
+            var textErrorFromComponent = Browser.Exists(By.Id("getTextError"), TimeSpan.FromSeconds(10));
+            Assert.NotNull(textErrorFromComponent);
+
+            var expectedError = "The incoming data stream of length 50000 exceeds the maximum allowed length 32000.";
+            Browser.Contains(expectedError, () => textErrorFromComponent.GetAttribute("innerHTML"));
+
+            FocusAway();
+            AssertLogDoesNotContainMessages(CircuitErrors);
+        }
+
+        [Fact]
+        public void CanEditValue_MinimalContent()
+        {
+            var textArea = Browser.Exists(By.Id("largeTextArea"), TimeSpan.FromSeconds(10));
+            Assert.NotNull(textArea);
+            textArea.SendKeys("abc");
+
+            Assert.Equal("abc", GetTextAreaValueFromBrowser());
+
+            FocusAway();
+            AssertLogDoesNotContainMessages(CircuitErrors);
+        }
+
+        [Fact]
+        public void CanEditValue_LargeAmountOfContent_Insert()
+        {
+            var newValue = new string('f', 50_000);
+            SetTextAreaValueInBrowser('f');
+
+            var textArea = Browser.Exists(By.Id("largeTextArea"), TimeSpan.FromSeconds(10));
+            Assert.NotNull(textArea);
+            textArea.SendKeys("abc");
+
+            Assert.Equal(newValue + "abc", GetTextAreaValueFromBrowser());
+
+            FocusAway();
+            AssertLogDoesNotContainMessages(CircuitErrors);
+        }
+
+        [Fact]
+        public async Task OnChangeTriggers()
+        {
+            var lastChangedTime = Browser.Exists(By.Id("lastChangedTime"));
+            Assert.NotNull(lastChangedTime);
+
+            var lastChangedLength = Browser.Exists(By.Id("lastChangedLength"));
+            Assert.NotNull(lastChangedLength);
+
+            var textArea = Browser.Exists(By.Id("largeTextArea"), TimeSpan.FromSeconds(10));
+            Assert.NotNull(textArea);
+            textArea.SendKeys("abc");
+            FocusAway();
+
+            var firstTick = Convert.ToInt64(lastChangedTime.GetAttribute("innerHTML"));
+            Assert.True(firstTick > 0);
+            var firstLength = Convert.ToInt32(lastChangedLength.GetAttribute("innerHTML"));
+            Assert.Equal(3, firstLength);
+
+            // Ensure time passes between first and second changes
+            await Task.Delay(1000);
```

Comment:
Is there something else we could do here? Adding waits tend to add unreliability and the fix usually is to increase the waits which just makes our tests slower.

Reply:
Removed the timing aspect entirely, realized it's not actually adding any additional value as we're able to validate a new event is triggered simply by verifying the `lastChangedLength` has increased after we insert more text.